src/uwtools/config/validator.py

```diff
@@ -57,6 +58,19 @@ def _prep_config(config: Union[dict, YAMLConfig, Optional[Path]]) -> YAMLConfig:
 def _validation_errors(config: Union[dict, list], schema: dict) -> List[str]:
     """
     Identify schema-validation errors.
+
+    :param config: A config to validate.
+    :param schema: JSON Schema to validate the config against.
+    :return: Any validation errors.
     """
-    validator = jsonschema.Draft202012Validator(schema)
+
+    # See https://github.com/python-jsonschema/referencing/issues/61 about typing issues.
+
+    def retrieve(uri: str) -> Resource:
+        name = uri.split(":")[-1]
+        with open(resource_path(f"jsonschema/{name}.jsonschema"), "r", encoding="utf-8") as f:
+            return Resource(contents=json.load(f), specification=DRAFT202012)  # type: ignore
+
+    registry = Registry(retrieve=retrieve)  # type: ignore
+    validator = jsonschema.Draft202012Validator(schema, registry=registry)
```

Comment:
The updated logic in this function configures `jsonschema` to dynamically load appropriate `.jsonschema` files from `uwtools.resources.jsonschema` when they are referenced by another schema, supporting DRYing out common elements between schemas -- for example, blocks describing component-execution properties (`execution.jsonschema`) or block describing namelists and their sets of names and values (`namelist.jsonschema`). `$ref` entries in schemas can now refer to other schemas as `urn:uwtools:<name>` (e.g. `urn:uwtools:execution`) to access the appropriate schema. See below for examples.